samples: intel_s1000_crb: sample application for digital microphones

sathishkuttan opened the pull request:
Sample application to receive audio from digital microphones in an Intel S1000 CRB.
Uses digital microphone driver APIs defined in `include/audio/dmic.h`

Commit:
soc: intel_s1000: add DMIC power control

Add DMIC power on to SoC init sequence

Signed-off-by: Sathish Kuttan <sathish.k.kuttan@intel.com>

Commit:
samples: intel_s1000_crb: add DMIC sample app

Add a sample application to demonstrate usage of digital microphones
in an Intel S1000 CRB

Signed-off-by: Sathish Kuttan <sathish.k.kuttan@intel.com>

rgundi commented on soc/xtensa/intel_s1000/soc.c:
```diff
@@ -186,13 +186,26 @@ u32_t soc_get_ref_clk_freq(void)
 		SOC_PWRCTL_DISABLE_PWR_GATING_DSP0;
+
+#if (CONFIG_AUDIO_INTEL_DMIC)
+	/* enable power */
```
You can move this section of code (202 to 207) to 193. That will ensure you use a single CONFIG_AUDIO_INTEL_DMIC which is more readable.

sathishkuttan replied:
Done.

rgundi reviewed (commented):
As you rightly pointed out, this sample needs a bit of documentation to elucidate how the DMIC is getting tested here.

Commit:
samples: intel_s1000_crb: set BOARD for dmic sample

Set BOARD variable in CMakeLists.txt for DMIC Sample App
Since this sample app is for intel_s1000_crb, setting BOARD
in CMakeLists.txt will eliminate the need to specify in
cmake command line every time during build.

Signed-off-by: Sathish Kuttan <sathish.k.kuttan@intel.com>

Commit:
samples: intel_s1000_crb: add doc for dmic sample

Add short documentation on sample app for DMIC driver usage
on Intel S1000 CRB.

Signed-off-by: Sathish Kuttan <sathish.k.kuttan@intel.com>

sathishkuttan commented:
> As you rightly pointed out, this sample needs a bit of documentation to elucidate how the DMIC is getting tested here.

Documentation added in commit 16903178ba7e0ab5b2b969b256e6b427d473296b

dbkinder commented on samples/boards/intel_s1000_crb/dmic/dmic.rst:
```diff
@@ -0,0 +1,46 @@
+   [00:00:00.990,004] <inf> dmic_sample.dmic_sample_app: Iteration 4/4 complete, 100 audio frames received.
+   [00:00:00.990,005] <inf> dmic_sample.dmic_sample_app: Exiting DMIC sample app ...
+
+  ...
```
Indent the ``...`` the same as the contents of the code block (to the ``[`` on the previous lines)

sathishkuttan replied:
done.

dbkinder commented on samples/boards/intel_s1000_crb/dmic/dmic.rst:
```diff
@@ -0,0 +1,46 @@
+Requirements
+************
+
+This application uses an Intel® S1000 Customer Reference Board (CRB)
```
Is there a board available to the public they can use to run this sample?

sathishkuttan replied:
Available only under NDA

dbkinder commented on samples/boards/intel_s1000_crb/dmic/dmic.rst:
```diff
@@ -0,0 +1,46 @@
+Overview
+********
+
+A sample implementation of an application that uses the digital microphone
```
Use a complete sentence:  
```
This sample application uses a digital microphone driver
to receive interleaved audio samples from a microphone array.

sathishkuttan replied:
Done

dbkinder commented on samples/boards/intel_s1000_crb/intel_s1000_crb.rst:
```diff
@@ -0,0 +1,10 @@
+.. _intel_s1000_crb-samples:
+
+Intel S1000 Customer Reference Board Samples
```
Since this is plural ("samples") will there be more coming, and is the "Customer Reference Board" available to the public?

sathishkuttan replied:
Yes. Board availability only under NDA.

sathishkuttan commented:
@dbkinder @rgundi - check for resolution to your comments and approve.
